### DISCUSSION TITLE
[Q&A] Additional content replacing parts of the original content

### DISCUSSION THREAD

**Discussion by dbeniamine** [glpi-project/glpi-agent#809] [Q&A] [answered]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

Hello,

I want to replace a part of the inventory when we run `glpi-agent`, with additional-content I know how to add a line, but is there a way to tell `glpi-agent` that my additional-content should replace the initial discovered content ?

**Comment by g-bougard** [comment]
Hello @dbeniamine 

you have to use [additional-content](https://glpi-agent.readthedocs.io/en/1.11/configuration.html#additional-content) option. It permits to load a XML or a json file depending on the inventory format. GLPI 10 expects json format and I'll focus on this case.

In json format, you will be able to merge datas where in XML you had to replace all the datas. If a data is still set in the before-merge generated inventory, the data from the json should simply override the discovered data.

Let's play with an example. If I generate a partial inventory for my own computer, the bios section looks like:
```
$ glpi-inventory --partial bios
{
   "action": "inventory",
   "content": {
      "bios": {
         "bdate": "2019-09-09",
         "bmanufacturer": "American Megatrends Inc.",
         "bversion": "1201",
         "mmanufacturer": "ASUSTeK COMPUTER INC.",
         "mmodel": "PRIME X570-P"
      },
...
```

Now I can use the following content as json `additional-content` option file:
```
{
   "content": {
      "bios": {
         "bdate": "2029-09-09"
      }
   }
}
```
To change the bios date:
```
$ glpi-inventory --partial bios --additional-content /tmp/bios-update.json
{
   "action": "inventory",
   "content": {
      "bios": {
         "bdate": "2029-09-09",
         "bmanufacturer": "American Megatrends Inc.",
         "bversion": "1201",
         "mmanufacturer": "ASUSTeK COMPUTER INC.",
         "mmodel": "PRIME X570-P"
      },
...
```

If you want to change all the section, you have to set a new value for all known values. To erase a value, set it to an empty string if this is a string.

Have you another case where this principle is not working as you wish ?

**Comment by dbeniamine** [comment]
Hello @g-bougard ,

Thanks for your quick answer, I am trying to replace the `remote_mgmt` section which is an array instead of an object and I think this is the issue for me.

By default I have something like

```json
"remote_mgmt" : [
   {
     "id": "someid",
     "type": "rustdesk"
   }
]
```

If I add an `additional-content` file like :  

```json
"remote_mgmt" : [
   {
     "id": "someid/?password=somerandompassword",
    "type": "rustdesk"
   }
]
```

I end up with : 

```json
"remote_mgmt" : [
   {
     "id": "someid",
    "type": "rustdesk"
   },
   {
     "id": "someid/?password=somerandompassword",
    "type": "rustdesk"
   }
]
```

While I would like to obtain : 

```json
"remote_mgmt" : [
   {
     "id": "someid/?password=somerandompassword",
    "type": "rustdesk"
   }
]
```

**Reply by dbeniamine** [reply, marked as answer]
Another approach could be simply to disable the rustdesk discovery in the glpi-agent settings thus the only `remote_mgmt` entry would be the one in the `additional-content` file .

Actually `glpi-agent --json --no-category=remote_mgmt --addition-content=myfile.json` seems to be doing the job, so in the end, adding `no-category=remote_mgmt`  to my `/etc/glpi-agent/conf.d/00-install.cf` does exactly what I want
